tests/jax/integration/parameter_server_test.py

```diff
@@ -1,89 +1,357 @@
-            executor_adder=mocks.MockAdder,
-            executor_environment_loop=mocks.MockExecutorEnvironmentLoop,
-            networks=mocks.MockNetworks,
-            trainer=mocks.MockTrainer,
-            trainer_dataset=mocks.MockTrainerDataset,
-            distributor=mocks.MockDistributor,
-        )
-        return components, {}
+from mava.systems.jax import System
+from tests.jax.integration.mock_systems import (
+    mock_system_multi_process,
+    mock_system_multi_thread,
+    mock_system_single_process,
+)
 
 
 @pytest.fixture
-def test_system() -> System:
-    """Dummy system with zero components."""
-    return TestSystem()
-
-
-def test_parameter_server_process_instantiate(
-    test_system: System,
-) -> None:
-    """Test if the parameter server instantiates processes as expected."""
-    # Environment.
-    environment_factory = functools.partial(
-        debugging_utils.make_environment,
-        env_name="simple_spread",
-        action_space="discrete",
-    )
+def test_system_sp() -> System:
+    """A single process built system"""
+    return mock_system_single_process()
 
-    # Networks.
-    network_factory = ippo.make_default_networks
 
-    test_system.build(
-        environment_factory=environment_factory,
-        network_factory=network_factory,
-        non_blocking_sleep_seconds=0,
-    )
+@pytest.fixture
+def test_system_mp() -> System:
+    """A multi process built system"""
+    return mock_system_multi_process()
+
+
+@pytest.fixture
+def test_system_mt() -> System:
+    """A multi thread built system"""
+    return mock_system_multi_thread()
+
+
+def test_parameter_server_single_process(test_system_sp: System) -> None:
+    """Test if the paraameter server instantiates processes as expected."""
     (
         data_server,
         parameter_server,
         executor,
         evaluator,
         trainer,
-    ) = test_system._builder.store.system_build
-    assert type(parameter_server) == ParameterServer
+    ) = test_system_sp._builder.store.system_build
+
+    # System parameter server
+    assert type(parameter_server.store.system_checkpointer) == savers.Checkpointer
+
+    for network in parameter_server.store.parameters["networks-network_agent"].values():
+        assert list(network.keys()) == ["w", "b"]
+        assert jnp.size(network["w"]) != 0
+        assert jnp.size(network["b"]) != 0
+
+    param_without_net = parameter_server.store.parameters.copy()
+    del param_without_net["networks-network_agent"]
+    assert param_without_net == {
+        "trainer_steps": jnp.zeros(1, dtype=jnp.int32),
+        "trainer_walltime": jnp.zeros(1, dtype=jnp.float32),
+        "evaluator_steps": jnp.zeros(1, dtype=jnp.int32),
+        "evaluator_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_steps": jnp.zeros(1, dtype=jnp.int32),
+    }
+
+    ############################get_parameters test#####################################
```

Comment:
as discussed, think we need to make this more high-level and not test these methods that were covered in the unit test for the parameter server